packages/compiler/react/transform.ts

```diff
@@ -227,12 +227,12 @@ export const transformComponent = (
 
   // We want to add a __props property for the original call props
   // TODO: refactor this probably
-  dynamics.data.push({
-    id: t.identifier('__props'),
-    value: params?.length
-      ? (params[0] as t.Expression)
-      : t.objectExpression([]),
-  });
+  if (params?.length) {
```

Comment:
```suggestion
  if (params?.length && t.isExpression(params[0])) {
```
[nits] this would be safer?